tests/SAML2/AuthnRequestTest.php

```diff
@@ -68,4 +71,42 @@ public function testExtensionOrdering()
 
         $this->assertXmlStringEqualsXmlString($document->C14N(), $authnRequest->toUnsignedXML()->C14N());
     }
+
+    public function testThatTheSubjectIsCorrectlyRead()
+    {
+        $document = new DOMDocument();
+        $document->loadXML(<<<AUTHNREQUEST
+<samlp:AuthnRequest
+    xmlns:samlp="urn:oasis:names:tc:SAML:2.0:protocol"
+    xmlns:saml="urn:oasis:names:tc:SAML:2.0:assertion"
+    AssertionConsumerServiceIndex="1"
+    Destination="https://tiqr.stepup.org/idp/profile/saml2/Redirect/SSO"
+    ID="_2b0226190ca1c22de6f66e85f5c95158"
+    IssueInstant="2014-09-22T13:42:00Z"
+    Version="2.0">
+  <saml:Issuer>https://gateway.stepup.org/saml20/sp/metadata</saml:Issuer>
+  <saml:Subject>
+        <saml:NameID Format="urn:oasis:names:tc:SAML:1.1:nameid-format:unspecified">user@example.org</saml:NameID>
+  </saml:Subject>
+</samlp:AuthnRequest>
+AUTHNREQUEST
+);
+        $authnRequest = new SAML2_AuthnRequest($document->documentElement);
+
+        $expectedNameId = array(
+            'Value'  => "user@example.org",
+            'Format' => "urn:oasis:names:tc:SAML:1.1:nameid-format:unspecified"
+        );
+        $this->assertEquals($expectedNameId, $authnRequest->getNameId());
+    }
+
+    public function testThatTheSubjectCanBeSetBySettingTheNameId()
```

Comment:
I'm missing a test for encrypted nameIds, would that be difficult to add?
Likewise for a simple test without a subject or nameidpolicy (as those make up the bulk of our current usage I'd like to have a test guaranteeing that they continue to work).